Shell/main.cpp

```diff
@@ -528,6 +528,36 @@ static int run_command(const String& cmd)
     return return_value;
 }
 
+CFile get_history_file()
+{
+    StringBuilder sb;
+    sb.append(g.home);
+    sb.append("/.history");
+    CFile f(sb.to_string());
+    if (!f.open(CIODevice::ReadWrite))
+        perror("history file");
```

Comment:
`perror()` will print whatever is in `errno`. There's no guarantee that it's relevant to the failed `open` call.
Instead, use `CFile::error_string()` to find the error message that relates to the failure.